benchmarking/example_classification.py

```diff
@@ -54,76 +71,88 @@
-        MACCSKeysFingerprint,
-        ERGFingerprint,
-        MAP4Fingerprint,
-        MHFPFingerprint,
-    ]
-
-    clf_names = ["RF", "LogReg", "LGBM"]
-    classifiers = [
-        RandomForestClassifier,
-        LogisticRegression,
-        lgb.LGBMClassifier,
-    ]
-    classifier_kwargs = [
-        {"n_jobs": -1, "class_weight": "balanced"},
-        {
-            "class_weight": "balanced",
-            "penalty": None,
-            "n_jobs": -1,
-        },
-        {"n_jobs": -1, "class_weight": "balanced", "verbose": 0},
-    ]
+    records = []
 
     np.random.seed(42)
 
-    for fingerprint, fp_name in zip(fprints, fp_names):
+    for fingerprint in fingerprint_classes:
+        fp_name = fingerprint.__name__
         records.append({})
         records[-1]["fp_name"] = fp_name
-        print(fp_name, "Fingerprint")
+        print(fp_name)
         start = time()
         fp_transformer = fingerprint(n_jobs=-1)
-        X_fp_train = fp_transformer.transform(X_train)
-        X_fp_valid = fp_transformer.transform(X_valid)
-        X_fp_test = fp_transformer.transform(X_test)
 
-        scaler = MinMaxScaler()
-        X_fp_train = scaler.fit_transform(X_fp_train)
-        X_fp_valid = scaler.fit_transform(X_fp_valid)
-        X_fp_test = scaler.transform(X_fp_test)
+        try:
+            X_fp_train = fp_transformer.fit_transform(X_train)
+            X_fp_valid = fp_transformer.transform(X_valid)
+            X_fp_test = fp_transformer.transform(X_test)
+        except RuntimeError as e:
+            print(f" - Error: {e}")
+            records[-1]["error"] = str(e)
+            continue
+        except ValueError as e:
+            print(f" - Error: {e}")
+            if conformers_generated:
+                try:
+                    X_fp_train = fp_transformer.transform(X_train_conf)
+                    X_fp_valid = fp_transformer.transform(X_valid_conf)
+                    X_fp_test = fp_transformer.transform(X_test_conf)
+                except ValueError as e:
+                    print(e.__class__.__name__)
+                    print(f" - Error: {e}")
+                    records[-1]["error"] = str(e)
+                    continue
+            else:
+                continue
 
         end = time()
         execution_time = end - start
         print(f" - Time of fingerprints computing : {round(execution_time,2)}s")
         records[-1]["execution_time"] = execution_time
-        for classifier, clf_name, clf_kwargs in zip(
-            classifiers, clf_names, classifier_kwargs
-        ):
+
+        imputer = SimpleImputer(strategy="median")
+        X_fp_train = imputer.fit_transform(X_fp_train)
+        X_fp_valid = imputer.transform(X_fp_valid)
+        X_fp_test = imputer.transform(X_fp_test)
+
+        scaler = RobustScaler()
+        X_fp_train = scaler.fit_transform(X_fp_train)
+        X_fp_valid = scaler.transform(X_fp_valid)
+        X_fp_test = scaler.transform(X_fp_test)
```

Comment:
I think we should have explicitly do this only for descriptors, since for fingerprints `RobustScaler` (or basically any scaling) would destroy sparsity. Descriptors list:
```
[
AutocorrFingerprint,
GETAWAYFingerprint,
MordredFingerprint,
MORSEFingerprint,
RDFFingerprint,
WHIMFingerprint
]
```

Reply:
by _this_ you mean imputing and scaling or only scaling?